### PR TITLE
release of version 0.4

### PR DESCRIPTION
**Pull request by jor-**
Since version 0.3.1 the master branch received some significant improvements. I recommend to release the version 0.4 to make this improvements available over PyPI etc..

I updated the changelog and recommend to update the copyright/author section.

### CODE REVIEW

**Comment by anntzer** [comment]
Can you take care of updating the copyright/authors too?

Given the issues with `setup_requires` (https://github.com/scikit-sparse/scikit-sparse/pull/23#issuecomment-249634485 and link therein), I think we should just revert #21: it's not worth it...

File: doc/changes.rst [hunk 1]
```diff
@@ -3,6 +3,13 @@ Changes
 
 .. module:: sksparse.cholmod
 
+v0.4
+------
+  * Support for 64-bit indices (type long) is added.
+  * Ordering method for cholesky decomposition is now choseable.
+  * The exception handling is refined.
```
Comment (anntzer) [inline doc/changes.rst]: "64-bit indices (type long) are now supported."
"The ordering method for Cholesky decomposition is now choosable."
"Specific exceptions subclasses are now thrown for each error condition."

Comment (jor-) [inline doc/changes.rst]: I changed it.

**Comment by jor-** [comment]
I updating the copyright and added the authors.

**Comment by jor-** [comment]
I think the problems regarding Cython are solved. All test passed (without preinstalled Cython). So I would like to stay with #21.

File: setup.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #! /usr/bin/env python
 
+# Copyright (C) 2008-2016 The scikit-sparse developers
```
Comment (anntzer) [inline setup.py]: I think this should go to the bottom of the list and only be a (C) since 2016 (but IANAL).

File: doc/overview.rst [hunk 3]
```diff
@@ -67,3 +67,13 @@ Contact
 
 Post your suggestions and questions directly to our `bug tracker
 <https://github.com/scikit-sparse/scikit-sparse/issues>`_.
+
+Developers
```
Comment (jor-) [inline doc/overview.rst]: Okay, I added the missing developer and sorted the list in alphabetical order.

File: setup.py [hunk 4]
```diff
@@ -1,8 +1,6 @@
 #! /usr/bin/env python
 
-# Copyright (C) 2008 Cournapeau David <cournape@gmail.com>
-# Copyright (C) 2009 Nathaniel Smith <njs@pobox.com>
-# Copyright (C) 2016 Antony Lee <anntzer.lee@gmail.com>
+# Copyright (C) 2008-2016 The scikit-sparse developers
```
Comment (anntzer) [inline setup.py]: I meant that you need to keep the previous copyright lines and add this one last.

File: setup.py (hunk 2 above)
Comment (jor-) [inline setup.py]: I forgot the remove the old copyright entries. I have done it now. So it is just the following.
```
Copyright (C) 2008-2016 The scikit-sparse developers
```
I think this is the best way. Or what do you think?

File: setup.py (hunk 4 above)
Comment (jor-) [inline setup.py]: Don't you think that it is somehow double to mention some of the developers here and (all) again in ``overview.rst``?

Comment (anntzer) [inline setup.py]: I think it's standard to mention both in the source and in the docs.

Comment (jor-) [inline setup.py]: If you want to mention them twice I would do it in a consistent way.
```
# Copyright (C) 2008-2016 The scikit-sparse developers:
# Alex Grigorievskiy
# Antony Lee
# Dag Sverre Seljebotn
# Joscha Reimer
# Leon Barrett
# Nathaniel Smith
# Yuri
```
Maybe sorted chronological if this is easy to get from git or in alphabetical order.

Comment (anntzer) [inline setup.py]: Sure.

Comment (jor-) [inline setup.py]: Oh, I forgot to include him in the list.

Comment (jor-) [inline setup.py]: Now there should be an entire (and equal) list in both files. Please take a look.

**Comment by jor-** [comment]
Do you have any further comments? Should I squash the commits?

**Comment by anntzer** [comment]
That seems fine.

However I just remembered that there's something else I wanted to look into: currently, I need to upload manually built docs to PyPI instead of relying on readthedocs, because RTD seems unable to document methods defined in cython.  Specifically, compare http://pythonhosted.org/scikit-sparse/cholmod.html#top-level-functions and http://scikit-sparse.readthedocs.io/en/latest/cholmod.html#top-level-functions.

Can you have a look?

**Comment by jor-** [comment]
I squashed the commits.

I will not have time to look into that in the near future. Sorry about that.

**Comment by anntzer** [comment]
Let's see whether https://github.com/rtfd/readthedocs.org/issues/2446 can get resolved quickly.

**Comment by jor-** [comment]
@anntzer: Do you still have in mind to make a release for conda and pip? Your problem with the doc is solved, right? I hope, I do not annoy you, but it would be nice to have a new release. ;-)

**Comment by anntzer** [comment]
Do you have a pypi account?  (Then the pypi package will be yours too :-))
I don't know whether there's anything on conda right now.  I couldn't care less for Linux; if you manage to make the package compile on Windows that'd be pretty nice though...

**Comment by jor-** [comment]
Yes, my pypi account name is jore. If you will gave me the necessary rights, I am going to make a new release on pypi. There is also an release on conda managed by conda-forge (https://github.com/conda-forge/scikit-sparse-feedstock). I will also take a look on this.

**Comment by anntzer** [comment]
There you go.

**Comment by grlee77** [comment]
@jor- 
If you are interested in being a maintainer for the conda-forge feedstock that would be great.  I don't see a new release up on pypi yet, but we can link to the release here at github as well.

**Comment by grlee77** [comment]
went ahead and made the PR at conda-forge and it seems to be working.  will plan to merge it there tomorrow
https://github.com/conda-forge/scikit-sparse-feedstock/pull/3

**Comment by jor-** [comment]
@anntzer: Thank you for the access on pypi. Can you also give me the necessary rights on testpypi?

@grlee77: Thank you for updating the conda-forge feedstock. It would be nice if you make me a maintainer for it as well.

**Comment by anntzer** [comment]
done.

**Comment by jor-** [comment]
Thank you. I made a release on pypi.

**Pull request merged**